src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -737,6 +755,72 @@ private void TimerOnElapsed(object? sender, EventArgs e)
             _timer!.Enabled = true;
         }
 
+        internal void ResetAddress(Address address)
+        {
+            _accountCache.RemoveAccounts([address]);
+        }
+
+        private sealed class AccountCache : IAccountStateProvider
+        {
+            private readonly IAccountStateProvider _provider;
+            private readonly LruCache<AddressAsKey, AccountStruct>[] _caches;
```

Comment:
why are there multiple cachess here?

Reply:
To reduce lock contention so it choose which bucket to go in (and therefore lock to use) based on last nibble of address so 16 caches